### PR TITLE
Spring Boot 2.3 update

### PR DESCRIPTION
**Pull request by murdos**
See https://github.com/jhipster/generator-jhipster/issues/11681

Associated `generator-jhipster` PR: https://github.com/jhipster/generator-jhipster/pull/11682

### CODE REVIEW

**Comment by DanielFran** [comment]
@murdos Would you mind to update reflection if mongobee is removed?

**Comment by murdos** [comment]
> 
> 
> @murdos Would you mind to update reflection if mongobee is removed?

I'm not sure to understand what you mean by "reflection". If you're talking about dependency management, yes, I was planning to do once Mongock provides a BOM.

**Comment by DanielFran** [comment]
Yes, I was refering the depency. Thanks

**Comment by DanielFran** [comment]
@murdos Do you think the branch 2.3 should be ready soon? Just to understand if I wait for the next updates of dependencies or do it after you push this branch to master...

**Comment by murdos** [comment]
> @murdos Do you think the branch 2.3 should be ready soon? Just to understand if I wait for the next updates of dependencies and do it after you push this branch to master...

Soon, I hope. The last remaining task is handling Couchbase upgrade.

**Comment by pascalgrimaud** [comment]
@murdos @DanielFran : do you think we can have Spring Boot 2.3 for the next release of JHipster 6.10.0 ?

**Comment by DanielFran** [comment]
@pascalgrimaud From last @murdos feedback, it is missing only Couchbase... but actually only @murdos is working on it.

And we have 3 other PR waiting for this merge: one for hazelcast from Vishal, another one for new probes and gracefull shutdown and a last one for dependencies update, bothe from me.

**Comment by murdos** [comment]
> 
> 
> @murdos @DanielFran : do you think we can have Spring Boot 2.3 for the next release of JHipster 6.10.0 ?

It depends... when do you want to release 6.10.0 ? 😉

> 
> @pascalgrimaud From last @murdos feedback, it is missing only Couchbase... but actually only @murdos is working on it.

I'm indeed working on Couchbase. And before the JHipster release, we'll need a release of Couchmove (https://github.com/differentway/couchmove/issues/10) (a snapshot is already available).

> And we have 3 other PR waiting for this merge: one for hazelcast from Vishal, another one for new probes and gracefull shutdown and a last one for dependencies update, bothe from me.

At least the one for hazelcast can wait for a later release, if needed, since it's an improvement.
Is your PR for probes and gracefull shutdown an improvement, or a fix, @DanielFran

**Comment by pascalgrimaud** [comment]
No release date yet, but the plan is:
- do a release of JHipster 6.10.0, including Spring Boot 2.3
- create a new branch for v7, from 6.10.0 -> so @avdev4j can start to work on this
- go on on 6.11.0, 6.12.0, etc if needed
- regularly merge main branch into v7 branch -> it will be @avdev4j's job :)

**Comment by DanielFran** [comment]
> Is your PR for probes and gracefull shutdown an improvement, or a fix, @DanielFran
@murdos It is a FIX since spring-boot 2.3 manage now correctly it defining 2 new endpoint for liveness and readiness

**Comment by murdos** [comment]
@DanielFran : with the merge of master in this branch, which includes breaking changes, you seem to assume that Spring Boot 2.3 will be released only in V7.
 Although this might be what would happen because of difficulties encountered with Couchbase, as far as I know, the decision has not been made yet to include Spring Boot 2.3 in a 6.x release or in V7

**Comment by DanielFran** [comment]
@murdos The idea was to have it in sync with master since it seems that the couchbase and mongock update will take some time.

We can always cherry pick the commit if needed to launch a new v6 with it included

File: jhipster-dependencies/pom.xml
Comment (DanielFran) [inline jhipster-dependencies/pom.xml:68]: @murdos version 4.1.14 is available

Comment (murdos) [reply jhipster-dependencies/pom.xml:68]: @DanielFran: we'll handle that after the merge. I really want to fix conflicts and merge the branch as soon as possible.

Comment (DanielFran) [reply jhipster-dependencies/pom.xml:68]: Agree, let's move on

**Comment by DanielFran** [comment]
@murdos Seems ready to be merged?

**Review by DanielFran — APPROVED** [review]
LGTM

**Comment by murdos** [comment]
> 
> 
> @murdos Seems ready to be merged?

Yes, it should! :D Let's just wait for CI tests

**Comment by DanielFran** [comment]
@murdos We have exactly same issues in https://github.com/jhipster/generator-jhipster/pull/12337/checks?check_run_id=1058663188, so I assume this is already broken!

File: .github/workflows/angular.yml
Comment (vishal423) [inline .github/workflows/angular.yml:36]: before merge, shouldn't this be reverted in all places?

File: pom.xml
Comment (vishal423) [inline pom.xml:19]: before merge, shouldn't this be reverted?

**Comment by mshima** [comment]
> @murdos We have exactly same issues in https://github.com/jhipster/generator-jhipster/pull/12337/checks?check_run_id=1058663188, so I assume this is already broken!

Looks like cypress triggered the failure.
The webflux session support is behaving differently from java.
Since webflux ci is broken before cypress was merged, I think we should go ahead and merge.

**Pull request merged**
